pkg/kubelet/kubelet_pods.go

```diff
@@ -159,14 +159,22 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 		if err != nil {
 			return nil, cleanupAction, err
 		}
-		if mount.SubPath != "" {
+
+		if mount.SubPathExpr != "" {
 			if !utilfeature.DefaultFeatureGate.Enabled(features.VolumeSubpath) {
 				return nil, cleanupAction, fmt.Errorf("volume subpaths are disabled")
 			}
 
-			// Expand subpath variables
-			if utilfeature.DefaultFeatureGate.Enabled(features.VolumeSubpathEnvExpansion) {
-				mount.SubPath = kubecontainer.ExpandContainerVolumeMounts(mount, expandEnvs)
+			if !utilfeature.DefaultFeatureGate.Enabled(features.VolumeSubpathEnvExpansion) {
+				return nil, cleanupAction, fmt.Errorf("volume subpath expansion is disabled")
+			}
+
+			mount.SubPath = kubecontainer.ExpandContainerVolumeMounts(mount, expandEnvs)
```

Comment:
we should not mutate the passed-in mount object (which is part of the passed-in pod object) here

Reply:
@kubernetes/sig-storage-api-reviews if the expanded subpath differs at different points in time, what implications does that have on the various stages of the volume lifecycle (e.g mount, remount, unmount, reconstruct, etc)?

Reply:
this is modifying the container object passed into makeMounts instead of a local variable. I'd expect something more like

```go
subPath := mount.SubPath
if len(mount.SubPathExpr) > 0 {
  subPath = kubecontainer.ExpandContainerVolumeMounts(mount, expandEnvs)
}

if len(subPath) > 0 {
  ...
}
```

Reply:
Okay. Thanks for clarifying.

Reply:
Used local variable

Reply:
> @kubernetes/sig-storage-api-reviews if the expanded subpath differs at different points in time, what implications does that have on the various stages of the volume lifecycle (e.g mount, remount, unmount, reconstruct, etc)? put another way, what in the volume handling code depends on mount paths (including subpaths) being immutable for a given pod?

Subpath value matters during mount (container creation).  But is not used during unmount or reconstruction.  I need to double check on remount.